### PR TITLE
docs: v1 to v2 upgrade guide

### PR DESCRIPTION
**Pull request by matifali**
Adds the upgrade guide for v1 to v2.

I could not find an easier way to preview and test the guide, but it should look like the recording below.
courtesy: https://registry.terraform.io/tools/doc-preview

https://github.com/user-attachments/assets/715bcfa5-a848-44ab-9a71-08fb442bb0c8

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -8,7 +8,11 @@ description: |-
 
 # Coder Provider
 
-Terraform provider for managing Coder [templates](https://coder.com/docs/templates), which are the underlying infrastructure for Coder [workspaces](https://coder.com/docs/workspaces).
+Terraform provider for managing Coder [templates](https://coder.com/docs/admin/templates), which are the underlying infrastructure for Coder [workspaces](https://coder.com/docs/user-guides/workspace-management).
```
Comment (matifali) [inline docs/index.md:11]: These links could be updated when we have a Glossary of terms.

File: docs/index.md [hunk 2]
```diff
@@ -8,7 +8,11 @@ description: |-
...
+
+-> Requires Coder v2.13.0 or later.
```
Comment (matifali) [inline docs/index.md:13]: We introduced a few attributes in the v2.13.0 release.

File: docs/guides/version-2-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    }
+  }
+}
+
+provider "coder" {}
+```
+
+## Provider Arguments
+
+Version 2.0.0 removes the [`feature_use_managed_variables`](https://registry.terraform.io/providers/coder/coder/1.0.4/docs#feature_use_managed_variables-1) argument from the `provider` block.
```
Comment (bpmct) [inline docs/guides/version-2-upgrade.md:61]: Maybe we should explain that this has been deprecated for a long time and has no impact on Coder workspace anymore.

**Review by bpmct — APPROVED** [review]
All LGTM!

**Review by johnstcn — APPROVED** [review]

File: docs/index.md (hunk 2 above)
Comment (johnstcn) [reply docs/index.md:13]: `coder_workspace_owner.login_type` is only available in v2.18.0 or later.

**Pull request merged**